use-cases/virgo/synthetic_data_gen/file_gen.py

```diff
@@ -0,0 +1,278 @@
+import numpy as np
+
+from tqdm.notebook import tqdm
+import h5py as h5
+import multiprocessing
+from tqdm import tqdm
+# import time
+from pathlib import Path
+
+import gwpy
+from gwpy.timeseries import TimeSeries
+
+if torch.cuda.is_available():
+    device = 'cuda'
+    print(torch.cuda.get_device_name(torch.cuda.current_device()))
+else:
+    device = 'cpu'
+    
+
+def cut_image(qplot_array,index_freq,index_time,square_size=64):
+    """
+    Cut qplot as square_size X square_size 2D np.array centered at peak frequency and corresponding time
+
+    Input:
+    - qplot_array (np.array) : qplot relative to the whole TimeSeries
+    - index_time (int) : time index in qtransform relative to peak frequency
+    - index_freq (int) : frequency index in qtransorm relative to peak frequency
+    - square_size (int) : Size in pixels of qplot image (default 64)
+
+    Return:
+    - subarray (np.array) : qplot cutted as square_size X square_size np.array
+
+    """
+
+    center_x = index_time
+    center_y = index_freq  # Replace with the actual y-coordinate
+
+
+    original_width, original_height = qplot_array.shape
+
+    # Calculate the starting and ending indices for the subarray
+    start_x = max(center_x - square_size // 2, 0)
+    #print(start_x)
+    end_x = min(start_x + square_size, original_width)
+    #print(end_x)
+    start_y = max(center_y - square_size // 2, 0)
+    # print(start_y)
+    end_y = min(start_y + square_size, original_height)
+    # print(end_y)
+    # start_y=0
+    # end_y=square_size
+
+
+    # Adjust indices if needed to make sure the resulting subarray is (square_size X square_size)
+    if end_x - start_x < square_size:
+        diff_x = square_size - (end_x - start_x)
+        if end_x < original_width:
+            end_x += diff_x
+        else:
+            start_x -= diff_x
+    if end_y - start_y < square_size:
+        diff_y = square_size - (end_y - start_y)
+        if end_y < original_height:
+            end_y += diff_y
+        else:
+            start_y -= diff_y
+
+    subarray = qplot_array[start_x:end_x, start_y:end_y]
+    # print(subarray.shape)
+    # print(type(subarray))
+
+    return subarray
+
+
+
+def extract_peak_frequency(hq):
+    """
+    Calculates peak frequency (and relative time) of a given qplot
+
+    Input:
+    -hq (gwpy.Spectrgram) : Qtransform
+
+    Return:
+    -index_time (int) : time index in qtransform relative to peak frequency
+    -index_freq (int) : frequency index in qtransorm relative to peak frequency
+    """
+
+    #Calculate peak frequency, time and energy density
+    index_flat = np.argmax(hq.value)
+
+    # Convert the flattened index to 2D index
+    index_time, index_freq = np.unravel_index(index_flat, hq.shape)
+    peak_freq = hq.frequencies.value[index_freq]#/converting_factor_frequency
+    peak_value=np.max(hq.value)
+    peak_time = hq.times.value[index_time]
+
+    return index_time,index_freq
+
+
+def process_image(row,row_idx,channels,square_size):
```

Comment:
Redefinition of function under `./use-cases/virgo/src/dataset.py::process_image`